### PR TITLE
Add ligature support to terminal

### PR DESCRIPTION
**Pull request by Tyriar**
Part of #34103

@deepak1556 see main/app

Todo:

- [ ] Finalize font access/chrome flag changes
- [ ] Fix disable ligatures without reload https://github.com/xtermjs/xterm.js/issues/3288
- [ ] Type ahead causing weird rendering https://github.com/xtermjs/xterm.js/issues/3288
- [ ] Support in web/fill in other get ctor functions
- [ ] Make sure font-finder isn't bundled into product build

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -144,7 +144,10 @@ function configureCommandlineSwitchesSync(cliArgs) {
 		'disable-color-correct-rendering',
 
 		// override for the color profile to use
-		'force-color-profile'
+		'force-color-profile',
+
+		// Enable experimental font access for terminal ligatures
+		`enable-features='FontAccess'`
```
Comment (deepak1556) [inline src/main.js:150]: Did this work ? I would expect appendSwitch to ignore this, it should instead be

`app.commandLine.appendSwitch('enable-features', 'FontAccess');`

Comment (Tyriar) [reply src/main.js:150]: Yep it worked, can change it though

Comment (deepak1556) [reply src/main.js:150]: lets keep it separate for consistency.

**Comment by Tyriar** [comment]
Closing in favor of https://github.com/microsoft/vscode/pull/139537

**Pull request closed without merging**